packages/roosterjs-editor-core/lib/corePlugins/CopyPlugin.ts

```diff
@@ -0,0 +1,59 @@
+import getColorNormalizedContent from '../darkMode/getColorNormalizedContent';
+import { Editor, EditorPlugin } from 'roosterjs-editor-core';
+
+/**
+ * Copy plugin, hijacks copy events to normalize the content to the clipboard.
+ */
+export default class CopyPlugin implements EditorPlugin {
+    private editor: Editor;
+    private copyDisposer: () => void;
+
+    /**
+     * Get a friendly name of  this plugin
+     */
+    getName() {
+        return 'Copy';
+    }
+
+    /**
+     * Initialize this plugin. This should only be called from Editor
+     * @param editor Editor instance
+     */
+    public initialize(editor: Editor) {
+        this.editor = editor;
+        this.copyDisposer = editor.addDomEventHandler('copy', this.onCopy);
+    }
+
+    /**
+     * Dispose this plugin
+     */
+    public dispose() {
+        this.copyDisposer();
+        this.copyDisposer = null;
+        this.editor = null;
+    }
+
+    private onCopy = (event: Event) => {
+        // if it's dark mode...
+        if (this.editor && this.editor.isDarkMode()) {
+            // get whatever the current selection range is
+            const selectionRange = this.editor.getSelectionRange();
+            if (selectionRange && !selectionRange.collapsed) {
+                const clipboardEvent = (event as ClipboardEvent);
```

Comment:
Suggest adding a check see if the copied content is just a single image, then skip everything below and let browser handle it.

Reply:
We actually can't, unfortunately. copy events don't allow you access to what's on the clipboard, only paste. However, if the copied content is just the image (right click copy image) then it'll skip this block anyway.